zerver/webhooks/jira/view.py

```diff
@@ -78,23 +81,59 @@ def convert_jira_markup(content: str, realm: Realm) -> str:
     full_link_re = re.compile(r"\[(?:(?P<title>[^|~]+)\|)(?P<url>[^\]]*)\]")
     content = re.sub(full_link_re, r"[\g<title>](\g<url>)", content)
 
-    # Try to convert a Jira user mention of format [~username] into a
-    # Zulip user mention. We don't know the email, just the Jira username,
-    # so we naively guess at their Zulip account using this
-    mention_re = re.compile(r"\[~(.*?)\]")
-    for username in mention_re.findall(content):
-        # Try to look up username
-        user_profile = guess_zulip_user_from_jira(username, realm)
-        if user_profile:
-            replacement = f"**{user_profile.full_name}**"
-        else:
-            replacement = f"**{username}**"
+    # We replace mentioned account [~accountid:60e5a86a471e61006a4c51fd]
+    # with their corresponding usernames fetched by Jira API.
+    url = get_url(payload)
+    if auth:
```

Comment:
Can this condition check for `url` too? if `get_url` returned `" "` then we can't make the API call too. I think that's worth putting for correctness, although in practice, the user's base URL looks like it's included in most payloads.